raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -1058,12 +1055,12 @@ contract TokenNetwork is Utils {
         returns (uint256, uint8)
     {
         Channel storage channel = channels[channel_identifier];
-        uint8 state = channel.state;
+        ChannelState memory state = channel.state;  // This must **not** update the storage
```

Comment:
why?

Reply:
as the compiler error says:
```
TokenNetwork.sol:1058:9: Error: Storage location can only be given for array or struct types.
        ChannelState memory state = channel.state;  // This must **not** update the storage
        ^-----------------------^
```
The default for local variables is `memory`. You can set them to `storage` only if they are `array`s or `struct`s. Otherwise, it gives you an error.

Reply:
The docs suggest this only applies for complex types: http://solidity.readthedocs.io/en/v0.4.24/types.html#data-location

Reply:
Hummm, I asusmed that `ChannelState state` would be a reference to the storage, and not a local variable. Thanks